server/src/test/java/org/elasticsearch/common/xcontent/support/filtering/AbstractXContentFilteringTestCase.java

```diff
@@ -47,8 +48,51 @@ private XContentBuilder createBuilder() throws IOException {
         return XContentBuilder.builder(getXContentType().xContent());
     }
 
-    private XContentBuilder createBuilder(Set<String> includes, Set<String> excludes) throws IOException {
-        return XContentBuilder.builder(getXContentType(), includes, excludes);
+    private XContentBuilder filter(Builder sample, Set<String> includes, Set<String> excludes) throws IOException {
+        if (randomBoolean()) {
+            return filterOnBuilder(sample, includes, excludes);
+        }
+        FilterPath[] excludesFilter = FilterPath.compile(excludes);
+        if (excludesFilter != null && Arrays.stream(excludesFilter).anyMatch(FilterPath::hasDoubleWildcard)) {
+            /*
+             * If there are any double wildcard filters the parser based
+             * filtering produced weird invalid json. Just field names
+             * and no objects?! Weird. Anyway, we can't use it.
+             */
+            return filterOnBuilder(sample, includes, excludes);
+        }
+        FilterPath[] includesFilter = FilterPath.compile(includes);
+        return filterOnParser(sample, includesFilter, excludesFilter);
```

Comment:
The tests that this enables are super comprehensive. It's kind of a shame to do it randomly though. We could probably use more subclasses or parametersized tests or something to run them all non-randomly. They are plenty fast.